src/Pages/WatchingRoom/WatchingRoom.jsx

```diff
@@ -88,6 +88,18 @@ function WatchingRoomPage() {
       }
       setVideoStatus("");
     });
+
+    newSocket.on(SOCKET_EVENTS.VIDEO_SKIPPED, ({ 
+      video: newVideo,
+      playlist: newPlaylist,
+    }) => {
+      setPlaylist(newPlaylist);
```

Comment:
```suggestion
      setPlaylist(...newPlaylist);
```
Asıl sorun burada, dizileri kopyalamak için `spread` operatörü kullanmamız gerekiyor.